templates/agent/src/components/ThemeSwitcher.tsx

```diff
@@ -0,0 +1,66 @@
+'use client'
+
+import {useTheme} from 'next-themes'
+import {useEffect, useState} from 'react'
+
+export function ThemeSwitcher() {
+	const {theme, setTheme} = useTheme()
+
+	const [mounted, setMounted] = useState(false)
+
+	useEffect(() => {
+		setMounted(true)
+	}, [])
+
+	if (!mounted) return null
+
+	return (
+		<div>
+			<button
+				className='bg-primary'
+				onClick={() => setTheme(theme === 'light' ? 'dark' : 'light')}>
+				{theme === 'light' ? (
+					<svg
+						xmlns='http://www.w3.org/2000/svg'
+						width='24'
+						height='24'
+						viewBox='0 0 24 24'
+						fill='none'
+						stroke='currentColor'
+						stroke-width='2'
+						stroke-linecap='round'
+						stroke-linejoin='round'
+						className='lucide lucide-moon'>
+						<path d='M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z' />
+					</svg>
```

Comment:
Could import these from `lucide-react`!

Reply:
ah, I just noticed that it's already there, was trying to avoid introducing new dependency, will amend, one moment